### PR TITLE
Allow analysis to fail

### PR DESCRIPTION
**Pull request by jasonLaster**
Should fix the crash we were seeing with invalid location. Of course, we should also still have a backend command error that we want to hunt down to explain why we were sending the bad location as well

### CODE REVIEW

File: src/ui/setup/helpers.ts [hunk 1]
```diff
@@ -55,6 +57,7 @@ export async function setupAppHelper(store: UIStore) {
     prefs,
     features,
     asyncStore,
+    debuggerAsyncStore,
```
Comment (bvaughn) [inline src/ui/setup/helpers.ts:60]: This seems unrelated to this PR. What is it for?

File: packages/protocol/analysisManager.ts [hunk 2]
```diff
@@ -160,6 +160,9 @@ class AnalysisManager {
         !handler.onAnalysisResult || client.Analysis.runAnalysis({ analysisId }, sessionId),
         !handler.onAnalysisPoints || client.Analysis.findAnalysisPoints({ analysisId }, sessionId),
       ]);
+    } catch (e) {
+      // log the error if the analysis fails
```
Comment (bvaughn) [inline packages/protocol/analysisManager.ts:164]: nit This comment is kind of useless 😁

File: packages/protocol/analysisManager.ts [hunk 3]
```diff
@@ -160,6 +160,9 @@ class AnalysisManager {
...
+      console.error(e);
```
Comment (bvaughn) [inline packages/protocol/analysisManager.ts:165]: Shouldn't this be calling `onAnalysisError` somehow? Otherwise won't it just hang forever?

**Review by bvaughn — COMMENTED** [review]
Not sure I understand this fix.

File: packages/protocol/analysisManager.ts (hunk 2 above)
Comment (jasonLaster) [reply packages/protocol/analysisManager.ts:164]: lol - so true...

File: packages/protocol/analysisManager.ts (hunk 3 above)
Comment (jasonLaster) [reply packages/protocol/analysisManager.ts:165]: Not sure, i think `onAnalysisError` is reserved for the response failing, as opposed to the request not being created... subtle difference.

Comment (bvaughn) [reply packages/protocol/analysisManager.ts:165]: Hm, interesting. So what's the expected behavior from the POV of the thing calling the API? It will just await the result and then it will need to handle the fact that neither complete nor error were called?

Comment (jasonLaster) [reply packages/protocol/analysisManager.ts:165]: Honestly, I gave this 10 minutes of thought earlier. My sole goal was don't crash. We should probably take another stab tomorrow

Comment (jcmorrow) [reply packages/protocol/analysisManager.ts:165]: If the analysis fails, I think we should call `onAnalysisError`, regardless of what the error is.

**Review by jcmorrow — APPROVED** [review]

File: src/ui/setup/helpers.ts (hunk 1 above)
Comment (bvaughn) [reply src/ui/setup/helpers.ts:60]: Was this change intentional to land? @jcmorrow

Comment (jcmorrow) [reply src/ui/setup/helpers.ts:60]: Oh, I totally forgot that this was even in here. It seems unrelated. I'll PR it back out.
